smithy-linters/src/main/java/software/amazon/smithy/linters/MissingSensitiveTraitValidator.java

```diff
@@ -0,0 +1,196 @@
+
+        public List<String> getPhrases() {
+            return phrases;
+        }
+
+        public void setPhrases(List<String> phrases) {
+            this.phrases = phrases;
+        }
+
+        public boolean getExcludeDefaults() {
+            return excludeDefaults;
+        }
+
+        public void setExcludeDefaults(boolean excludeDefaults) {
+            this.excludeDefaults = excludeDefaults;
+        }
+    }
+
+    private final Set<String> sensitiveWords;
+    private final Set<String> sensitivePhrases;
+
+    private MissingSensitiveTraitValidator(Config config) {
+        if (!config.getExcludeDefaults()) {
+            Set<String> phrasesInit = new HashSet<>(DEFAULT_SENSITIVE_PHRASES);
+            phrasesInit.addAll(config.getPhrases()
+                    .stream()
+                    .map(phrase -> phrase.toLowerCase(Locale.US))
+                    .collect(Collectors.toSet()));
+            sensitivePhrases = Collections.unmodifiableSet(phrasesInit);
+            sensitiveWords = Collections.unmodifiableSet(DEFAULT_SENSITIVE_WORDS);
+        } else {
+            if (config.getPhrases().isEmpty()) {
+                //This configuration combination makes the validator a no-op.
+                throw new IllegalArgumentException("Cannot set 'excludeDefaults' to true and leave "
+                                                 + "'phrases' empty or unspecified.");
+            }
+            sensitivePhrases = Collections.unmodifiableSet(new HashSet<>(config.getPhrases()));
+            sensitiveWords = Collections.EMPTY_SET;
+        }
+    }
+
+    /**
+     * Finds shapes without the sensitive trait that possibly contain sensitive data,
+     * based on the shape/member name and the list of key words and phrases.
+     * @param model Model to validate.
+     * @return list of violation events
+     */
+    @Override
+    public List<ValidationEvent> validate(Model model) {
+        List<ValidationEvent> validationEvents = new ArrayList<>();
+        validationEvents.addAll(scanShapeNames(model));
+        validationEvents.addAll(scanMemberNames(model));
+        return validationEvents;
+    }
+
+    private List<ValidationEvent> scanShapeNames(Model model) {
+        List<ValidationEvent> validationEvents = new ArrayList<>();
+        model.shapes()
+                .filter(shape -> !shape.isMemberShape()
+                        && !shape.isOperationShape()
+                        && !shape.isServiceShape()
+                        && !shape.isResourceShape())
+                .filter(shape -> !shape.hasTrait(SENSITIVE))
+                .filter(shape -> containsSensitiveWord(shape.toShapeId().getName())
+                        || containsSensitivePhrase(shape.toShapeId().getName()))
+                .forEach(shape -> validationEvents.add(emit(shape)));
+
+        return validationEvents;
+    }
+
+    private List<ValidationEvent> scanMemberNames(Model model) {
+        return model.shapes()
```

Comment:
Hmm, the reason I didn't go that route was that I need to filter out members where their enclosing shape is already marked sensitive, but the member's target is not. Otherwise we get false positives